mountpoint-s3-client/src/endpoint.rs

```diff
@@ -24,13 +22,24 @@ pub struct Endpoint {
 impl Endpoint {
     /// Create a new endpoint for the given S3 region. This method automatically resolves the right
     /// endpoint URI to target.
-    pub fn from_region(region: &str, addressing_style: AddressingStyle) -> Result<Self, EndpointError> {
-        if AWS_PARTITION_REGEX.is_match(region) {
-            // TODO: support partitions other than "aws"
-            Self::from_uri_inner(&format!("https://s3.{region}.amazonaws.com"), addressing_style)
-        } else {
-            Err(EndpointError::UnsupportedRegion(region.to_owned()))
-        }
+    pub fn from_region(
+        region: &str,
+        addressing_style: AddressingStyle,
+        allocator: &Allocator,
```

Comment:
don't think this needs an allocator argument, it should just use `Allocator::default()` like `S3CrtClient::new()` does.